### ISSUE TITLE
meson can't find dependency via cmake

### ISSUE THREAD

**Issue by yshui** [mesonbuild/meson#5605]
Digging through the log files, I found out cmake failed with:

```
CMake Error at /usr/share/cmake-3.14/Modules/CheckIncludeFile.cmake:93 (try_compile):
 Unknown extension ".c" for file

   [snip]/meson-private/cmake_prometheus-cpp/CMakeFiles/CMakeTmp/CheckIncludeFile.c

 try_compile() works only for enabled languages.  Currently these are:

   C CXX

 See project() command to enable other languages.
```

Which is a really absurd error.

Going into the mentioned directory then run cmake manually works.

meson version is 0.51.0

**Comment by mensinda** [comment]
Could you please attach the full log (via pastebin etc.) and the name of the dependency?

**Comment by yshui** [comment]
This is how you reproduce this problem: https://gist.github.com/yshui/f203c379f501cecf4923000bcecc804c

Note, if you comment out the `find_dependency(Threads)` line in the `.cmake`, this problem goes away.

@mensinda

**Comment by colatkinson** [comment]
I can confirm that this is also affecting me with meson 0.51.0 on Windows 10. I can provide logs if it would be helpful.

**Comment by mensinda** [comment]
Should be fixed with #5630.

**Comment by yshui** [comment]
Thanks!
